types/validator_set.go

```diff
@@ -1059,7 +1061,10 @@ func ValidatorSetFromProto(vp *tmproto.ValidatorSet) (*ValidatorSet, error) {
 
 	vals.Proposer = p
 
-	vals.totalVotingPower = vp.GetTotalVotingPower()
+	// NOTE: We can't trust the total voting power given to us by other peers. If someone were to
+	// inject a non-zeo value that wasn't the correct voting power we could assume a wrong total
+	// power. We should look to remove TotalVotingPower from proto or add it in the validators hash.
+	vals.totalVotingPower = 0
```

Comment:
and here...

Reply:
On fromProto, wouldn't you want to populate it? you could use your local validatorset power, right?

Reply:
talked in slack, this looks good. It should not be breaking.